addons/mail/models/ir_model.py

```diff
@@ -16,9 +16,14 @@ class IrModel(models.Model):
     )
 
     def unlink(self):
-        # Delete followers for models that will be unlinked.
+        # Delete followers, messages and activities for models that will be unlinked.
+        models = tuple(self.mapped('model'))
         query = "DELETE FROM mail_followers WHERE res_model IN %s"
-        self.env.cr.execute(query, [tuple(self.mapped('model'))])
+        self.env.cr.execute(query, [models])
+
+        self.env['mail.activity'].search([('res_model', 'in', models),]).unlink()
```

Comment:
is this necessary? cf. https://github.com/odoo/odoo/blob/12.0/addons/mail/models/mail_activity.py#L112

Reply:
letme verify...

Reply:
yup not required